### DISCUSSION TITLE
Github action does not run when multiple files were added/removed

### DISCUSSION THREAD

**Discussion by pedrovsg** [community/community#57830] [Actions]
### Select Topic Area

Bug

## Description:

I have encountered a bug in GitHub Actions where actions that run only when files are changed in a certain path do not run when many files are modified. This issue severely affects the functionality and reliability of workflows, especially in cases where required tests are not run.

## Steps to Reproduce:

1. Create a new repository or use an existing one.
2. Set up a workflow that includes multiple actions that perform on changes in a certain path.
`  pull_request:
    paths:
      - '.github/workflows/*'
      - 'src/main/**'`
4. Make sure the workflow is triggered when a push event occurs.
5. Modify a large number of files in the repository (e.g., more than 1000 files).
6. Push the changes to the repository.

## Expected Behavior:

The workflow should trigger and execute all the actions regardless of the number of files changed. 

## Actual Behavior:

When a large number of files are modified, the action does not execute.

## Additional Information:

This issue is more pronounced when the number of modified files exceeds a certain threshold, although the exact limit is uncertain.

## Workaround:

Currently, there is no reliable workaround to ensure the execution of actions on all modified files when a large number of files are changed. One possible workaround is to split the changes into smaller prs. However, this is an inconvenient and time-consuming solution, especially in cases where the workflow's success depends on processing all the changes together.

## Impact:

The bug significantly impacts developers who rely on GitHub Actions to automate their workflows. It hinders productivity, introduces inconsistencies, and undermines the reliability of the development process.

**Comment by make-ing** [comment]
We are currently experiencing the same behavior. We have a similar configuration where we want to trigger an action on pull-request with three paths set as filter. When i manually create a PR with a lot of new/changed files (more than 300 in this case) the action is not triggering.

Can some help?

P.S. In our special case we have only one file which is changed which matches one of the paths filters and the rest of the files are not matching any path filter.

**Comment by hwhsu1231** [comment]
Hello, GitHub Community.

I recently seem to have encountered the same/similar issue.

Related to: https://github.com/dorny/paths-filter/issues/227

Originally, I was using `on.pull_request.paths` in a GitHub Actions workflow to check if `l10n/**.pot` files had changed. However, I found that the workflow was not triggered. I then tried using `dorny/paths-filter` instead, and the result was the same. Later, I discovered the reason for the issue from the log output:

1. Files changed in the PR created by `peter-evans/create-pull-request@v6` are about **4128**.
2. Files changed detected by `dorny/paths-filter@v3` are only **3000**.

From the log output, it seems that:

The PR contains too many files changed, so that `dorny/paths-filter@v3` is unable to load all the files changed.

**Comment by pomSense** [comment]
+1 on this issue. We run a mono repo running on yarn workspaces and pnp. When we need to update our yarn version i.e. 3->4, there are thousands of file changes since yarn automatically updates cached modules during the upgrade.

This prevents our ci from running which uses `on` `pull_request` `paths`:
```yaml
on:
  pull_request:
    branches: [main]
    paths: # workflow will not RUN unless you make changes inside path folder
      - 'packages/core/core/**'
```

I suspect this is due to the paths have a search limit of some sort when checking for path changes. 

Is there a temp workaround where we can tell the action to NOT look in the `.cache` files when looking for paths?

Ideally I wish the file limit was larger, but when I updated yarn 3->4, looks like I broke github's file limit as it shows an infinity symbol:
<img width="1035" alt="image" src="https://github.com/community/community/assets/35952778/0a8a1725-64db-4f60-a3bb-be8b63946e39">

**Comment by jcjones1515** [comment]
I'm seeing this as well on PRs with large numbers of files

**Comment by thanos-kataras** [comment]
Very similar case, with fewer files
1. Changed ~700 files, 148 pipelines were executed
2. Reverted the changes to 350 files and 300 pipelines were executed
3. Identified one of those that were not executed, created a new PR with the same changes only for that file and it was picked up correctly.

More files present an even more erratic behavior. Seems like a pagination issue limiting the files that are being checked to just a subset of them.

**Comment by Mugane** [comment]
Sounds like nobody at GitHub cares

**Reply by Thexumaker** [reply]
They still don't care in 2025 xD

**Comment by Thexumaker** [comment]
Same here but in 2025

**Comment by GentileFulvio** [comment]
I notice the same issue

**Comment by SuperHiroki** [comment]
Same here.

https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax
> Diffs are limited to 300 files. If there are files changed that aren't matched in the first 300 files returned by the filter, the workflow will not run. You may need to create more specific filters so that the workflow will run automatically.
